### ISSUE TITLE
Possible data truncation in rfc822_base64() | Content boundary encoding issue

### ISSUE THREAD

**Issue by radarsymphony** [freescout-helpdesk/freescout#2026]
### Summary

An encoding error is created in the fetching logs for messages sent from a gmail/google account and with attachments. When Freescout encounters one of these messages, it creates an error message in the log for each sync/fetch attempt and prevents any messages received after from being synced from that account (or other accounts after it in the fetch queue without this patch https://github.com/freescout-helpdesk/freescout/commit/4bb44a8d1a4c95a9a486fc06b29ae8b5167649a3 in the system) until that message is removed/marked read.

### Steps to reproduce

Using `msmtp` and the message below, I was able to reproduce the issue on some runs.
1. On server with MTA configured, setup msmtp config.
2. Copy the message below into a text file on that server.
3. Update the addresses to reflect your system.
4. Run `msmtp -t < message.txt`

- Freescout v1.8.14
- PHP 7.3
- Patch mentioned above manually inserted into code

I will be upgrading to Freescout 1.8.16 and php 8.0 soon and will report back when I have re-tested.

<details><summary>Mime Message</summary>
<pre><code>
MIME-Version: 1.0
From: noreply@example.com
Subject: Test attachment encoding with boundary
To: freescout-inbox@example.com
Content-Type: multipart/mixed; 
	boundary="00000000000023421205e06a86af"

--00000000000023421205e06a86af
Content-Type: text/csv
Content-Disposition: attachment;
    filename="Test_file.csv"

key,value,
name,bob,

--00000000000023421205e06a86af
Content-Type: text/plain

Body of text goes here...

--00000000000023421205e06a86af
</code></pre>
</details>

### What is the current *bug* behavior?

Messages with certain content boundaries become stuck in the imap folder and are not fetched my freescout and prevent subsequent messages from being fetched until the offending message is removed.

### What is the expected *correct* behavior?

Freescout would  circumvent these issue emails or fetch them regardless of their 'boundary' format, which might currently be the case in different environment.

### Relevant logs and/or screenshots

A successfully fetched message may still show the boundary in the message body.

![image](https://user-images.githubusercontent.com/73671474/171743674-b30adb47-d4f2-47f7-8916-f9133a76ee27.png)

**Examples of boundary caused issue (from gmail)**
1. `Content-Type: multipart/mixed; boundary="00000000000023421205e06a86af"`
2. `Content-Type: multipart/mixed; boundary="000000000000835fa905e0032a6d"`

`Folder: INBOX; Error: Possible data truncation in rfc822_base64(): 00000000000023421205e06a86ae--`

`Folder: INBOX; Error: Possible data truncation in rfc822_base64(): 000000000000835fa805e0032a6c--`

| Instance | String |
| --- | ---|
| boundary 1 | 00000000000023421205e06a86a**f** |
| error 1 | 00000000000023421205e06a86a**e**-- |
| boundary 2 | 000000000000835fa**9**05e0032a6**d** |
| error 2 | 000000000000835fa**8**05e0032a6**c**-- |

**Example of boundary that did _not_ cause issue (from gmail)**
- `Content-Type: multipart/mixed; boundary="00000000000063598205e047f72a"`

### Possible fixes

I am not sure yet. The work around is to remove the message or mark it 'read' in the imap folder and re-fetch the rest of the messages.

This issue appears very similar to the one reported here https://github.com/freescout-helpdesk/freescout/issues/1980

**Comment by freescout-helpdesk** [comment]
Update FreeScout to the latest version.

**Comment by radarsymphony** [comment]
I have updated to 1.8.16 and PHP 8.0.18. The issue has not presented itself again naturally. Maybe the conditions haven't been right yet?

My test sequence (described above) still creates nearly the same behavior as before, but slightly different. Before, the message containing the incompatible boundary would not be fetched and messages after it would also not be fetched permanently. Now, the problematic message _is_ fetched while the messages after it ares (initially) not, but on the next sync cycle the remaining messages are fetched.

A section of a fetch log fetching only one message when several were available:
```
[2022-06-08 07:59:50] Folder: INBOX
[2022-06-08 07:59:50] Fetched: 1
[2022-06-08 07:59:50] 1) Test attachment encoding with boundary
[2022-06-08 07:59:50] Message-ID is empty, generated artificial Message-ID: fs-622941b4ca52fb5adbc9c61673a88bf5@[redacted]
[2022-06-08 07:59:50] Message from: Customer
[2022-06-08 07:59:50] Thread successfully created: 868442
[2022-06-08 07:59:50] Folder: Sent Items
[2022-06-08 07:59:50] Fetched: 0
[2022-06-08 07:59:50] Folder: Outbox
[2022-06-08 07:59:50] Fetched: 0
```
I'll keep monitoring for now. Maybe my test sequence does not recreate all of the factors necessary for the issue to occur.

**Comment by freescout-helpdesk** [comment]
Any news?

**Comment by radarsymphony** [comment]
Hi @freescout-helpdesk, 

I have successfully updated Freescout to 1.8.16 and php 8.0. The issue has not occurred naturally since.

However, I can still generate similar behavior with the replication steps I outlined above. The behavior of these steps has changed a bit since the upgrade, but is still resulting in partial fetches. Now, the problematic message is fetched, but the remaining messages behind it are not until the next fetch cycle. It seems like my test sequence isn't close enough to the real scenario and maybe the behavior I'm seeing from that testing is actually due to an unrelated reason. For example, as I've shown above, Freescout needing to generate a message ID disrupting the normal fetch routine: `Message-ID is empty, generated artificial Message-ID`

I can re-open this issue resurfaces due to natural operation.
